enterprise/frontend/src/metabase-enterprise/google_sheets/GoogleSheetManagement.tsx

```diff
@@ -0,0 +1,400 @@
+                {helperText}
+              </Text>
+            )}
+          </div>
+        </Flex>
+        {status === "complete" && (
+          <Button variant="subtle" onClick={handleClick}>
+            {t`Add New`}
+          </Button>
+        )}
+      </Flex>
+    </Menu.Item>
+  );
+}
+
+export function GsheetConnectionModal({
+  isModalOpen,
+  onClose,
+  reconnect,
+}: {
+  isModalOpen: boolean;
+  onClose: () => void;
+  reconnect: boolean;
+}) {
+  const gSheetsSetting = useSetting("gsheets");
+  const userIsAdmin = useSelector(getUserIsAdmin);
+  const { data: { email: serviceAccountEmail } = {} } =
+    useGetServiceAccountQuery();
+
+  const gSheetsEnabled = useSetting("show-google-sheets-integration");
+
+  if (
+    !FIXME_TEST_BOOLEAN &&
+    (!gSheetsEnabled || !gSheetsSetting || !userIsAdmin || !serviceAccountEmail)
+  ) {
+    console.error("Google Sheets integration is not enabled");
+    return null;
+  }
+
+  const { status, folder_url } = gSheetsSetting;
+
+  return (
+    isModalOpen &&
+    (status === "not-connected" ? (
+      <GoogleSheetsConnectModal
+        onClose={onClose}
+        serviceAccountEmail={serviceAccountEmail}
+        folderUrl={folder_url}
+      />
+    ) : (
+      <GoogleSheetsDisconnectModal onClose={onClose} reconnect={reconnect} />
+    ))
+  );
+}
+
+const ModalWrapper = ({
+  children,
+  onClose,
+}: {
+  children: React.ReactNode;
+  onClose: () => void;
+}) => (
+  <Modal opened onClose={onClose} size="lg">
+    <Flex px="lg" pb="lg" gap="md" direction="column">
+      {children}
+    </Flex>
+  </Modal>
+);
+
+function GoogleSheetsConnectModal({
+  onClose,
+  folderUrl,
+  serviceAccountEmail,
+}: {
+  onClose: () => void;
+  folderUrl: string | null;
+  serviceAccountEmail: string;
+}) {
+  const dispatch = useDispatch();
+  const [folderLink, setFolderLink] = useState(folderUrl ?? "");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const [saveFolderLink, { isLoading: isSavingFolderLink }] =
+    useSaveGsheetsFolderLinkMutation();
+
+  const onSave = async (event: FormEvent) => {
+    event.preventDefault();
+    setErrorMessage("");
+
+    const validationRegex = /(https|http)\:\/\/drive\.google\.com\/.+/;
+
+    if (!validationRegex.test(folderLink.trim())) {
+      setErrorMessage(t`Invalid Google Drive folder link`);
+      return;
+    }
+
+    trackSheetImportClick();
+    await saveFolderLink({
+      url: folderLink.trim(),
+    })
+      .unwrap()
+      .then(() => {
+        dispatch(reloadSettings());
+        onClose();
+      })
+      .catch(response => {
+        setErrorMessage(response?.data?.message ?? "Something went wrong");
+      });
+  };
+
+  return (
+    <ModalWrapper onClose={onClose}>
+      <Text size="lg" fw="bold">
+        {
+          // eslint-disable-next-line no-literal-metabase-strings -- admin only string
+          t`Share the Google Drive folder that contains your Google Sheets with Metabase`
```

Comment:
Maybe: `Share with Metabase the Google Drive folder that contains your Google Sheets`?